contracts/settling_game/L05_Wonders.cairo

```diff
@@ -0,0 +1,280 @@
+    # stop here if over latest claimable epoch
+    let ( local within_max_epoch ) = is_nn_le(claiming_epoch, current_epoch - 1) 
+    if within_max_epoch == 0:
+        return()
+    end
+
+    let ( controller ) = controller_address.read()
+    let ( wonders_state_address ) = IModuleController.get_module_address(contract_address=controller, module_id=9)
+
+    let ( epoch_upkept ) = IS05_Wonders.get_wonder_epoch_upkeep(wonders_state_address, current_epoch, token_id)
+    if epoch_upkept == 1:
+        let (treasury_address) = IModuleController.get_treasury_address(contract_address=controller)
+        let ( resources_address ) = IModuleController.get_resources_address(contract_address=controller)
+        let ( epoch_total_wonders ) = IS05_Wonders.get_total_wonders_staked(contract_address=wonders_state_address, epoch=claiming_epoch)
+
+        let ( ids_arr ) = alloc()
+        let ( amounts_arr ) = alloc()
+
+        # Get claimable resources
+        loop_resources_claim(
+            wonders_state_address,
+            token_id,
+            current_epoch,
+            epoch_total_wonders,
+            1,
+            ids_arr,
+            1,
+            amounts_arr)
+
+        # Transfer claimable resources
+        IERC1155.safeBatchTransferFrom(
+            resources_address,
+            caller,
+            treasury_address,
+            22, 
+            ids_arr, 
+            22, 
+            amounts_arr)
+    else:
+
+    # Recurse
+    return loop_epochs_claim(
+        caller,
+        token_id,
+        current_epoch,
+        claiming_epoch + 1)
+    end
+    return ()
+end
+
+func loop_resources_claim{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+            wonders_state_address : felt, token_id : Uint256, epoch : felt, epoch_total_wonders : felt,
+            resource_claim_ids_len : felt, 
+            resource_claim_ids : felt*, 
+            resource_claim_amounts_len : felt, 
+            resource_claim_amounts : felt*) 
+        -> (
+            resource_claim_ids_len : felt, 
+            resource_claim_ids : felt*, 
+            resource_claim_amounts_len : felt, 
+            resource_claim_amounts : felt*):
+    alloc_locals
+
+    let ( below_max_id ) = is_nn_le(resource_claim_ids_len, 22)
+
+    assert_in_range(resource_claim_amounts_len, 1, 2**15)
+    if below_max_id == 1:
+        let ( resource_pool ) = IS05_Wonders.get_tax_pool(wonders_state_address, epoch, resource_claim_ids_len)
+        assert resource_claim_ids[resource_claim_ids_len - 1] = resource_claim_ids_len
+        assert resource_claim_amounts[resource_claim_ids_len - 1] = resource_pool / epoch_total_wonders 
+    
+        return loop_resources_claim(
+            wonders_state_address, token_id, epoch, epoch_total_wonders,
+            resource_claim_ids_len + 1, 
+            resource_claim_ids, 
+            resource_claim_amounts_len + 1, 
+            resource_claim_amounts)
+    else:
+    return (
```

Comment:
The indent here is weird as well. But since you're returning from the `if` branch, you could just replace the `else` with `end` and keep the rest.